com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/ModelUtilTest.java

```diff
@@ -482,4 +486,35 @@ public void testInvalidFile() throws Exception {
         Modelimporter.Scene scene = loadBuiltScene("broken.gltf", meshSetBuilder, animSetBuilder, skeletonBuilder);
         assertTrue(scene == null);
     }
+
+    @Test
+    public void testVehicleGltfHierarchy() throws Exception {
+        Rig.MeshSet.Builder meshSetBuilder = Rig.MeshSet.newBuilder();
+        Modelimporter.Scene scene = loadBuiltScene("vehicle.glb", meshSetBuilder);
+        
+        // Validate scene loaded successfully
+        assertNotNull("Vehicle scene should load", scene);
+        
+        // Get all models from the meshset
+        List<Rig.Model> models = meshSetBuilder.getModelsList();
+        
+        // Check for duplicate models by collecting IDs
+        Set<Long> modelIds = new HashSet<>();
+        
+        for (Rig.Model model : models) {
+            long id = model.getId();
+            String name = "Model_" + id; // Since we hash the node name
+            
+            assertFalse("Model ID " + id + " should be unique (no duplicates)", 
```

Comment:
in tested model `Group` was added twice
<img width="550" height="776" alt="CleanShot 2025-08-15 at 10 26 10@2x" src="https://github.com/user-attachments/assets/7a0a1b91-f91a-42d8-b8cc-4073ddb4927e" />